source/tests/system/nirfmxbt_driver_api_tests.cpp

```diff
@@ -104,6 +125,62 @@ TEST_F(NiRFmxBTDriverApiTests, InitializeFromNIRFSA_Close_Succeeds)
+  EXPECT_SUCCESS(session, client::cfg_rf(stub(), session, "", 1e9, 0.0, 0.0));
+  EXPECT_SUCCESS(session, client::cfg_iq_power_edge_trigger(stub(), session, "", "0", IQPowerEdgeTriggerSlope::IQ_POWER_EDGE_TRIGGER_SLOPE_RISING_SLOPE, -20.0, 0.0, TriggerMinimumQuietTimeMode::TRIGGER_MINIMUM_QUIET_TIME_MODE_AUTO, 100e-6, IQPowerEdgeTriggerLevelType::IQ_POWER_EDGE_TRIGGER_LEVEL_TYPE_RELATIVE, Boolean::BOOLEAN_TRUE));
+  EXPECT_SUCCESS(session, client::cfg_packet_type(stub(), session, "", PacketType::PACKET_TYPE_DH1));
+  EXPECT_SUCCESS(session, client::cfg_data_rate(stub(), session, "", 1000000));
+  EXPECT_SUCCESS(session, client::cfg_payload_length(stub(), session, "", PayloadLengthMode::PAYLOAD_LENGTH_MODE_AUTO, 10));
+  EXPECT_SUCCESS(session, client::select_measurements(stub(), session, "", MeasurementTypes::MEASUREMENT_TYPES_ACP, Boolean::BOOLEAN_TRUE));
+  EXPECT_SUCCESS(session, client::acp_cfg_burst_synchronization_type(stub(), session, "", AcpBurstSynchronizationType::ACP_BURST_SYNCHRONIZATION_TYPE_PREAMBLE));
+  EXPECT_SUCCESS(session, client::acp_cfg_averaging(stub(), session, "", AcpAveragingEnabled::ACP_AVERAGING_ENABLED_FALSE, 10));
+  AcpOffsetChannelMode offsetChannelMode = AcpOffsetChannelMode::ACP_OFFSET_CHANNEL_MODE_SYMMETRIC;
+  EXPECT_SUCCESS(session, client::acp_cfg_offset_channel_mode(stub(), session, "", offsetChannelMode));
+  if (offsetChannelMode == AcpOffsetChannelMode::ACP_OFFSET_CHANNEL_MODE_SYMMETRIC)
+  {
+    EXPECT_SUCCESS(session, client::acp_cfg_number_of_offsets(stub(), session, "", 5));
+  }
+  else if (offsetChannelMode == AcpOffsetChannelMode::ACP_OFFSET_CHANNEL_MODE_INBAND)
+  {
+    EXPECT_SUCCESS(session, client::cfg_channel_number(stub(), session, "", 0));
+  }
+  EXPECT_SUCCESS(session, client::initiate(stub(), session, "", ""));
+
+  auto referenceChannelPower = client::acp_fetch_reference_channel_power(stub(), session, "", 10.0);
+  ni::tests::system::EXPECT_SUCCESS(referenceChannelPower);
+  auto measurementResponse = client::acp_fetch_offset_measurement_array(stub(), session, "", 10.0);
+  ni::tests::system::EXPECT_SUCCESS(measurementResponse);
+  auto maskTrace = client::acp_fetch_mask_trace(stub(), session, "", 10.0);
+  ni::tests::system::EXPECT_SUCCESS(maskTrace);
+  auto absolutePowerTrace = client::acp_fetch_absolute_power_trace(stub(), session, "", 10.0);
+  ni::tests::system::EXPECT_SUCCESS(absolutePowerTrace);
+  auto fetchedSpectrum = client::acp_fetch_spectrum(stub(), session, "", 10.0);
+  ni::tests::system::EXPECT_SUCCESS(fetchedSpectrum);
+
+  EXPECT_GT(referenceChannelPower.reference_channel_power(), 0.0);
+  EXPECT_THAT(measurementResponse.lower_absolute_power(), Each(Ne(0.0)));
+  EXPECT_THAT(measurementResponse.upper_absolute_power(), Each(Ne(0.0)));
+  EXPECT_THAT(measurementResponse.lower_relative_power(), Each(Ne(0.0)));
+  EXPECT_THAT(measurementResponse.upper_relative_power(), Each(Ne(0.0)));
+  EXPECT_GT(measurementResponse.actual_array_size(), 0.0);
```

Comment:
nit: I think all of the size comparisons should be GT `0` (integer value) not `0.0` (float value).

Reply:
Why compare float to int?

Reply:
I think he's just talking about the assertions on array sizes, all of which should be ints.

Reply:
Ah, right.

Reply:
Yeah, bad lack of emphasis. The ***SIZE*** comparisons should be ints. Everything else is correct as floats (I think).

Reply:
Changed the array size comparisons to use an integer value.